dread.js

```diff
@@ -12,20 +13,37 @@ function code(v) {
     return err => err.code === v;
 }
 
+const JitterType = Object.freeze({
+	NONE: "none",// use full backoff
+	FULL: "full",// random value between 0 and backoff
+	HALF: "half",// random value between backoff/2 and backoff
+});
+
 function exp(options = {}) {
-    const { base = 100, factor = 2, limit = 10000, jitter = true } = options;
+    const { base = 100, factor = 2, limit = 10000, jitter = JitterType.FULL} = options;
 
     assert(Number.isInteger(base), 'base must be an integer');
     assert(Number.isInteger(factor), 'factor must be an integer');
-    assert(Number.isInteger(limit), 'limit must be an integer');
-    assert(typeof jitter === 'boolean', 'jitter must be an boolean');
-
-    const cap = limit || Number.MAX_SAFE_INTEGER;
-    const amount = number => Math.min(base * Math.pow(factor, number), cap);
-
-    return jitter
-        ? number => Math.round(base + Math.random() * (amount(number) - base))// jitters between base and amount(number)
-        : number => Math.round(amount(number));
+	assert(Number.isInteger(limit), 'limit must be an integer');
+	assert(typeof jitter !== 'boolean', `jitter type boolean is deprecated, use JitterType ${util.inspect(JitterType)}`);
```

Comment:
let's just treat true/false as FULL ?

Reply:
Yeah was considering that couldn't decide.